next/src/field/type.ts

```diff
@@ -1,47 +1,130 @@
 import type { JsfSchemaType } from '../types'
 
-/**
- * WIP type for UI field output that allows for all `x-jsf-presentation` properties to be splatted
- * TODO/QUESTION: what are the required fields for a field? what are the things we want to deprecate, if any?
- */
-export interface Field {
-  name: string
-  label?: string
-  description?: string
-  fields?: Field[]
-  // @deprecated in favor of inputType,
+export type FieldType = 'text' | 'number' | 'select' | 'file' | 'radio' | 'group-array' | 'email' | 'date' | 'checkbox' | 'fieldset' | 'money' | 'country' | 'textarea'
+
+interface BaseField {
   type: FieldType
-  inputType: FieldType
+  name: string
+  label: string
   required: boolean
+  inputType: FieldType
   jsonType: JsfSchemaType
-  isVisible: boolean
-  accept?: string
-  errorMessage?: Record<string, string>
   computedAttributes?: Record<string, unknown>
+  errorMessage: Record<string, string>
+  schema: any
+  isVisible: boolean
+  description?: string
+  statement?: {
+    title: string
+    inputType: 'statement'
+    severity: 'warning' | 'error' | 'info'
+  }
+  [key: string]: unknown
```

Comment:
This line _hides_ a lot of TS errors that are hard to address without it. I'll keep investigating...